service/src/main/java/org/cbioportal/service/impl/SampleServiceImpl.java

```diff
@@ -173,11 +174,15 @@ private void processSamples(List<Sample> samples, String projection) {
 
         if (projection.equals("DETAILED")) {
             Map<String, Set<String>> sequencedSampleIdsMap = new HashMap<>();
+            Map<String, Set<String>> structuralVariantSampleIdsMap = new HashMap<>();
             List<String> distinctStudyIds = samples.stream().map(Sample::getCancerStudyIdentifier).distinct()
                 .collect(Collectors.toList());
+            
             for (String studyId : distinctStudyIds) {
                 sequencedSampleIdsMap.put(studyId,
                                           new HashSet<String>(sampleListRepository.getAllSampleIdsInSampleList(studyId + SEQUENCED)));
+                structuralVariantSampleIdsMap.put(studyId,
+                                       new HashSet<String>(sampleListRepository.getAllSampleIdsInSampleList(studyId + STRUCTURAL_VARIANT)));
```

Comment:
Note for future performance improvements : this seems like a good opportunity for caching ... otherwise every call to the sample service for a DETAILED response is recomputing these lists. I wonder if we could just add a function call in sampleListRepository which returned a HashSet rather than a list. (then it would be cached)

Reply:
good point! we should add a hashSet when we switch to the structural variant table.

Reply:
very good point, @sheridancbio! Let's have a follow up PR on this once we switch to the structural variant table. Thanks!